### PR TITLE
Allow admin to remove user from event

### PR DESCRIPTION
**Pull request by xuebingli**
@zendesk/volunteer-eng

This PR allows an admin to removes users from event and supersedes #43. Built on top of #43, this PR includes refactoring and two bug fixes that

- Fix bug with user avatar rendering
- Fix bug that a wrong user is removed

### CODE REVIEW

**Comment by xuebingli** [comment]
cc @craig-day @thebritican

File: app/javascript/components/NamedAvatar.js [hunk 1]
```diff
@@ -16,17 +20,79 @@ const styles = {
...
+          togglePopover('removeUserFromEvent', { userId })
+        }}
+      >
+        Remove
+      </a>
+    )}
+    {present(popover) &&
+      popover.data.userId === userId && (
+        <Dialog
+          title={'Remove User ' + name + ' from Event'}
```
Comment (craig-day) [inline app/javascript/components/NamedAvatar.js]: can use template strings here 

```jsx
title={`Remove User ${name} from Event`}
```

I would also maybe just drop the word "User" from the thing so it reads "Remove hello from Event"

File: app/javascript/components/NamedAvatar.js [hunk 2]
```diff
@@ -16,17 +20,79 @@ const styles = {
   name: {
     fontWeight: 600,
   },
-  subtitle: {},
+  remove: {
+    marginLeft: '20px',
```
Comment (craig-day) [inline app/javascript/components/NamedAvatar.js]: `px` is the default. You can just use the integer 20 here.

File: app/javascript/components/NamedAvatar.js [hunk 3]
```diff
@@ -16,17 +20,79 @@ const styles = {
...
+    cursor: 'pointer',
+  },
+  dialogActionsLink: {
+    padding: '15px',
+    cursor: 'pointer',
+    fontSize: '16px',
```
Comment (craig-day) [inline app/javascript/components/NamedAvatar.js]: same as above

File: app/javascript/components/NamedAvatar.js [hunk 4]
```diff
@@ -16,17 +20,79 @@ const styles = {
...
+    )}
+    {present(popover) &&
+      popover.data.userId === userId && (
+        <Dialog
+          title={'Remove User ' + name + ' from Event'}
+          actions={dialogActions(togglePopover, onRemove)}
+          open
+          onRequestClose={() => {
+            togglePopover('removeUserFromEvent')
+          }}
```
Comment (craig-day) [inline app/javascript/components/NamedAvatar.js]: any reason for the extra braces here? Can't it just be 

```jsx
onRequestClose={() => togglePopover()}
```

File: app/javascript/components/NamedAvatar.js [hunk 5]
```diff
@@ -16,17 +20,79 @@ const styles = {
...
+          actionsContainerStyle={{ paddingBottom: 20, textAlign: 'center' }}
```
Comment (craig-day) [inline app/javascript/components/NamedAvatar.js]: can we move these styles to the `styles` hash for the file?

File: app/javascript/pages/admin/Admin/index.js [hunk 6]
```diff
@@ -87,24 +87,15 @@ const OfficeFilter = ({ adminOfficeFilter, changeAdminOfficeFilter, offices }) =
...
 
   render() {
-    const {
-      routing,
-      children,
-      adminOfficeFilter,
-      changeAdminOfficeFilter,
-      data: { currentUser, offices },
-    } = this.props
+    const { routing, children, adminOfficeFilter, changeAdminOfficeFilter, data: { currentUser, offices } } = this.props
```
Comment (craig-day) [inline app/javascript/pages/admin/Admin/index.js]: did prettier single line this? The line seems pretty long.

File: app/javascript/pages/admin/Events/edit.js [hunk 7]
```diff
@@ -11,15 +11,23 @@ import Loading from 'components/LoadingIcon'
 
 import EventQuery from './queries/show.gql'
 import UpdateEventMutation from './mutations/update.gql'
+import DestroySignupMutation from 'mutations/DestroySignupMutation.gql'
```
Comment (craig-day) [inline app/javascript/pages/admin/Events/edit.js]: does this need to be a relative like the other ones?

**Review by craig-day — CHANGES_REQUESTED** [review]
Now that we are using the `present` helper in so many places, should we call it `isPresent`?

File: app/javascript/pages/admin/Admin/index.js (hunk 6 above)
Comment (xuebingli) [reply app/javascript/pages/admin/Admin/index.js]: Yes, prettier did this. I could commit with `--no-verify` if this line is annoying and/or put up a separate PR to decrease the max line length setting of the linter.

**Comment by xuebingli** [comment]
@craig-day bump

File: app/javascript/pages/admin/Admin/index.js (hunk 6 above)
Comment (craig-day) [reply app/javascript/pages/admin/Admin/index.js]: nah, I trust our prettier config. It was just surprised.

**Review by craig-day — CHANGES_REQUESTED** [review]
All of this code looks good, but lets add a check to ensure that the user executing the `destroySignup` mutation is either destroying their own signup, or is an admin. 

https://github.com/zendesk/volunteer_portal/blob/593c7b1309e916d1379e5a4ec53a375477af3ed9/app/graph/types/mutation_graph_type.rb#L23-L27

You should have the current user from the context, which is the third argument to the resolver.

**Comment by xuebingli** [comment]
@craig-day thanks for reviewing the PR. I've addressed the comment.

**Review by craig-day — APPROVED** [review]

**Pull request merged**